pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -642,6 +642,44 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 			cdLog.Debug("not creating new provision since the deployment is set to try install only once")
 			return reconcile.Result{}, nil
 		}
+
+		if cd.Spec.LimitInstallRestarts != nil && cd.Status.InstallRestarts > *cd.Spec.LimitInstallRestarts {
```

Comment:
I think that this should be greater than **or equal to**. The `InstallRestarts` field is the number of restarts that have already been done.
```suggestion
		if cd.Spec.LimitInstallRestarts != nil && cd.Status.InstallRestarts >= *cd.Spec.LimitInstallRestarts {
```

Reply:
`The InstallRestarts field is the number of restarts that have already been done.` This statement is not true. We first increment InstallRestarts when previous provision fails and then go ahead and create new Provision.